### PR TITLE
fix(pg-cdc): fix support for partitioned table

### PR DESCRIPTION
**Pull request by KeXiangWang**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

The bug is found by user.
The bug contains two symptom:
1. If the specified `publication` is created without `publish_via_partition_root = true`, the source/table creation will throw an error with message `ERROR: relation "original_table" is already member of publication "rw_publication"`.
2. If the `publication` is created by RW, RW cannot get any data.

The root cause is the `publish_via_partition_root = true`. Only with this option, the wal of each parition will be published in the root table's publication.

The fix is inspired by https://groups.google.com/g/debezium/c/NroAVr2saxU/m/eSeS8hYXCwAJ

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [x] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

https://github.com/risingwavelabs/risingwave-docs/pull/2628

</details>

### CODE REVIEW

**Comment by KeXiangWang** [comment]
The fix in this PR works. However, due to a [bug in sea-schema](https://github.com/SeaQL/sea-orm/issues/2115), CI fails. Their fix is [here](https://github.com/SeaQL/sea-schema/pull/125), which is included in `0.15.0`. However, the upgrading to `0.15.0` of `sea-schema` is non-trivial. Will create another PR to fix.

File: java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/connector/source/common/PostgresValidator.java [hunk 1]
```diff
@@ -370,6 +372,31 @@ private void validatePublicationConfig(boolean isSuperUser) throws SQLException
...
+            // make sure the publication are created with `publish_via_partition_root = true`, which
+            // is required by partitioned tables.
+            throw ValidatorUtils.invalidArgument(
+                    "Table '"
+                            + tableName
+                            + "' has partitions, which requires publication '"
+                            + pubName
+                            + "' to be created with `publish_via_partition_root = true`. \nHint: you can run `SELECT pubviaroot from pg_publication WHERE pubname = '"
+                            + pubName
+                            + "'` to check.");
```
Comment (StrikeW) [inline java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/connector/source/common/PostgresValidator.java:397]: nits: `... on upstream to check.`

**Review by StrikeW — APPROVED** [review]
LGTM!

File: java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/connector/source/common/PostgresValidator.java (hunk 1 above)
Comment (KeXiangWang) [reply java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/connector/source/common/PostgresValidator.java]: done

**Pull request merged**